lib/airbrake/worker.ex

```diff
@@ -110,9 +120,12 @@ defmodule Airbrake.Worker do
     unless ignore?(exception) do
       enhanced_options = build_options(options)
       payload = Airbrake.Payload.new(exception, stacktrace, enhanced_options)
-      json_encoder = Application.get_env(:airbrake_client, :json_encoder, Poison)
-      @http_adapter.post(notify_url(), json_encoder.encode!(payload), @request_headers)
+      json_payload = Airbrake.JSONEncoder.encode!(payload)
+      @http_adapter.post(notify_url(), json_payload, @request_headers)
     end
+  rescue
+    UndefinedFunctionError ->
```

Comment:
Is there any chance that we could hit an UndefinedFunctionError here at this rescue that isn't necessarily from `encode!/1` but some other undefined function?

Reply:
`@http_adapter` could be set to a module that doesn't work, but someone would have to use an unadvertised `config` option or find a way to not depend on `httpoison`.  The latter is _really_ hard since `httpoison` is a non-optional dependency of `airbrake_client`.

Someone would have to go out of there way to break that call.